### ISSUE TITLE
Chest Cart with Hopper Below Causes Exception Ticking Block Entity

### ISSUE THREAD

**Issue by immijimmi** [Railcraft/Railcraft#2016]
**Description of the Bug**
With a chest cart placed on a track on top of a hopper, placing items into the chest cart causes the world to crash and continue crashing when trying to start back up.

**To Reproduce**
1. Place at least 3 hoppers leading sideways into 3 furnaces, in a row
2. Place track on hoppers forming a line across all 3 hoppers
3. Place an additional 3 hoppers leading down into each furnace
4. Replace one track with a boost track and place a redstone torch next to it
5. Place a chest cart on one of the regular tracks
6. Open chest cart and place coal inside

**Expected behavior**
Coal is moved into the hopper below the chest cart and from the hopper into the bottom slot of the furnace

**Screenshots & Video**
![image](https://user-images.githubusercontent.com/9651027/119955748-434d3680-bf98-11eb-94eb-142a13d84cb2.png)

**Logs & Environment**
- Log: https://pastebin.com/YE9h2qhd
 - Crash report: https://pastebin.com/AznRMwn4

**Additional context**
This was reproduced successfully once in the server environment that the crash originated in, but could not then be replicated in single player via the above steps. Unsure if this means the bug only happens on a server or if there was a particularly esoteric detail that we missed.

Extra details that may be relevant:
- The server is running the [Polka Dot Door modpack](https://www.curseforge.com/minecraft/modpacks/polka-dot-door), on a Magma server with additional plugins (although none of the plugins are expected to have contributed to the crash)
- The crash occurred in Aroma1997's mining dimension
- A Buildcraft quarry was running in the vicinity (but not interacting with the setup that caused the crash)
- Big Reactors adds a recipe for coal to be crafted into graphite bars via a furnace, meaning hoppers will place coal into the top slot of the furnace if the hopper is placed above the furnace

**Comment by CovertJaguar** [comment]
Yeah... I don't think I do anything that could cause a difference from vanilla behavior here.
